src/VisualStudio/Core/Def/Implementation/Workspace/Esent/EsentPersistentStorage.cs

```diff
@@ -274,12 +274,22 @@ private bool TryGetUniqueId(string value, bool fileCheck, out int id)
 
             try
             {
+                // check whether value already exist to reduce allocation
+                if (_nameTableCache.TryGetValue(value, out id))
```

Comment:
ya, we can do that, but I dont think we need to do anything here. the allocation is something we can't avoid. the allocation reported here is inclusive allocation meaning it includes all allocations under that lambda. unless we can reduce number of times we execute the lambda, there is not that much allocations to save here.

I thought we are creating lambda unnecessarily, but as @jasonmalinowski pointed out, we already have check to remove those cases (except when race happens), and I don't think race is the major contributor here.